design/secrets_decoding_design.md

```diff
@@ -0,0 +1,241 @@
+  - user: policy/path/base64_user     # valid annotation with content-type
+    content-type: base64
+  - policy/path/base64_password:      # this is also valid when no alias is needed
+      content-type: base64
+```
+
+P2F already leverages Kubernetes' Downward API and volumeMounts to pass these annotations into the Secrets Provider container. This will make it possible to determine which secrets should be decoded before pushing them to the Kubernetes secret file. 
+
+## Design
+The goal of this design is to make usage of this feature as consistent as possible across different configurations. Due to the nature of configuring Conjur secrets in K8s secrets mode versus P2F mode, it makes the most sense to add the configuration in-place where Conjur secrets are already defined.
+* Kubernetes Secrets mode - add content-type to the existing K8s secrets manifest
+* P2F mode - add content-type to the existing pod annotations
+
+In both modes the solution will need to be backwards compatible with existing annotation formats. 
+
+The logic and structs should added for this feature should be shared between both modes when possible. For example, P2F mode uses [SecretSpec struct](https://github.com/cyberark/secrets-provider-for-k8s/tree/main/pkg/secrets/common/secret_spec.go) which could be updated with a ContentType field. Ideally both YAML formats can be unmarshaled into a common struct since they will contain the same data:
+* Alias
+* Path
+* ContentType
+  * Allowed values = ['text', 'base64']
+  * Default = 'text'
+
+### Kubernetes Secrets Mode
+In order to keep Conjur secrets configuration from creeping into other locations, we will add the content-type under the existing `conjur-map` annotation of the secrets manifest. This requires adding an additional supported multiline format:
+```
+conjur-map: |-
+  <alias_value>: <path_value>                 # supported
+  <alias_value>:                              # to be added
+    path: <path_value>
+    content-type: <content_type_value>    
+```
+
+In the case where multiple Kubernetes secrets reference the same Conjur secret, we should support a different content-type definition for each instance of the secret. i.e. 'k8s-secret-decode' can reference `path/to/somevar` with a content-type of base64, and `k8s-secret-dont-decode` can reference `path/to/somevar` with a content-type of text, and each K8s secret will be updated with the correct decoded or non-decoded secret value.
+
+### P2F Mode
+For P2F, we can update the existing `conjur.org/conjur-secrets.{secret-group}` annotations with the following allowed formats:
+```
+conjur.org/conjur-secrets.groupname: |
+  - <alias_value>: <path_value>               # supported
+  - <path_value>                              # supported
+  - <alias_value>: <path_value>               # to be added
+    content-type: <content_type_value>
+  - <path_value>:                             # to be added
+      content-type: <content_type_value>
```

Comment:
For this case with the implicit key the user will have to add a colon to the line, and the indenting is different that the case with the alias is specified which could lead to user errors. So since they would have to either add a colon on the end, or add the alias_value at the beginning, I think it's clearer to just support the one case which is the ` - <alias_value>: <path_value>` and add the content-type.

Reply:
Agreed and updated in the doc